src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs

```diff
@@ -203,6 +205,52 @@ string SaveRandomConfig()
             Assert.Equal(dynamicConfigMessage2, config["Hello"]); // Config DID reload from disk
         }
 
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34580", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        public async Task CreateDefaultBuilder_SecretsDoesReload()
+        {
+            var secretId = Assembly.GetExecutingAssembly().GetName().Name;
+            var reloadFlagConfig = new Dictionary<string, string>() { { "hostbuilder:reloadConfigOnChange", "true" } };
+            var secretPath = PathHelper.GetSecretsPathFromSecretsId(secretId);
+
+            var secretFileInfo = new FileInfo(secretPath);
+            Directory.CreateDirectory(secretFileInfo.Directory.FullName);
+
+            string SaveRandomSecret()
+            {
+                var newMessage = $"Hello ASP.NET Core: {Guid.NewGuid():N}";
+                File.WriteAllText(secretPath, $"{{ \"Hello\": \"{newMessage}\" }}");
+                return newMessage;
+            }
+
+            string dynamicSecretMessage1 = SaveRandomSecret();
+
+            var host = Host.CreateDefaultBuilder(new[] { "environment=Development", $"applicationName={secretId}" })
+                .ConfigureHostConfiguration(builder =>
+                {
+                    builder.AddInMemoryCollection(reloadFlagConfig);
+                })
+                .Build();
+
+            var config = host.Services.GetRequiredService<IConfiguration>();
+
+            Assert.Equal(dynamicSecretMessage1, config["Hello"]);
+
+            string dynamicSecretMessage2 = SaveRandomSecret();
+
+            var configReloadedCancelTokenSource = new CancellationTokenSource();
+            var configReloadedCancelToken = configReloadedCancelTokenSource.Token;
+
+            config.GetReloadToken().RegisterChangeCallback(o =>
+            {
+                configReloadedCancelTokenSource.Cancel();
+            }, null);
+            // Wait for up to 1 minute, if config reloads at any time, cancel the wait.
+            await Task.WhenAny(Task.Delay(TimeSpan.FromMinutes(1), configReloadedCancelToken)); // Task.WhenAny ignores the task throwing on cancellation.
```

Comment:
One more thought, would we want this test to be flagged [outerloop](https://github.com/dotnet/runtime/blob/master/docs/project/writing-tests.md#outerloop) since this adds delay?

Reply:
I am not sure about this, so decision is up to you. If you decide to mark it with `[OuterLoop]`, make sense to mark two other tests with delay too and to do it in separated PR

Reply:
The test above this does the same thing. The delay is only a "timeout" in case the `configReloadedCancelToken` never fires. The test will take much less time than a minute to execute under normal circumstances.